docs/groovy-script/mods/selectionguicrafting/recipe.md

```diff
@@ -0,0 +1,255 @@
+
+Don't know what a builder is? Check [the builder info page](../../getting_started/builder.md) out.
+
+:::::::::: details mods.selectionguicrafting.recipe.recipeBuilder() {open id="abstract"}
+- `Integer`. The amount of XP that will be granted to the player when the recipe is crafted. Requires greater than or equal to 0. (Default `0`).
+
+    ```groovy:no-line-numbers
+    xp(int)
+    ```
+
+- `Integer`. The time in ticks it takes to craft the recipe. Requires greater than or equal to 0. (Default `0`).
+
+    ```groovy:no-line-numbers
+    time(int)
+    ```
+
+- `ArrayList<GsTool>`. The tool that will be used to craft the recipe. Tool refers to the mainhand item. This doesn't need to be an item with durability. The first `float` value is the multiplier for the durability consumption. The second `float` value is the multiplier for the crafting time. Requires greater than or equal to 1.
+
+    ```groovy:no-line-numbers
+    tool(IIngredient)
+    tool(IIngredient, float)
+    tool(IIngredient, float, float)
+    ```
+
+- `ResourceLocation`. The path to the frame texture. Overrides the category frame. [Textures that come with the mod](https://github.com/Ender-Development/selection-gui-crafting-continued/tree/master/src/main/resources/assets/selectionguicrafting/textures/gui/frame). (Default `selectionguicrafting:textures/gui/frame/default.png`).
+
+    ```groovy:no-line-numbers
+    frame(String)
+    frame(ResourceLocation)
+    ```
+
+- `ArrayList<Ingredient>`. The input items that are required to craft the recipe. Input refers to the offhand item. You can add multiple inputs each with a different amount. Requires greater than or equal to 1.
+
+    ```groovy:no-line-numbers
+    input(IIngredient)
+    ```
+
+- `ArrayList<GsOutput>`. The output items that will be given to the player. You can add multiple outputs each with a different chance. The `float` value is the chance of the item being dropped. Requires greater than or equal to 1.
+
+    ```groovy:no-line-numbers
+    output(ItemStack)
+    output(ItemStack, float)
+    ```
+
+- `ArrayList<GsSound>`. The sounds that will be played when the recipe is crafted. Overrides the category sounds. The first `float` value is the volume, the second `float` value is the pitch of the sound.
+
+    ```groovy:no-line-numbers
+    sound(String, float, float)
+    sound(SoundEvent, float, float)
+    sound(ResourceLocation, float, float)
+    ```
+
+- `GsCatalyst`. The catalyst that will be used to craft the recipe. The catalyst is an optional additional input that must be present in the player's inventory. The `float` value is the chance of the catalyst being consumed. Requires greater than or equal to 0 and less than or equal to 1.
+
+    ```groovy:no-line-numbers
+    catalyst(IIngredient)
+    catalyst(IIngredient, float)
+    ```
+
+- `String`. The category that the recipe belongs to. If the category doesn't exist, it will throw an error. Requires not null.
```

Comment:
in the `@Comp` annotation, add `UNI = "lang_key"` with the key being something like `the category must exist`, this should generate `Requires not null and the category must exist.`